src/Hypergrid.js

```diff
@@ -1550,6 +1550,20 @@ Hypergrid.prototype = {
         });
         this.canvas.dispatchEvent(clickEvent);
     },
+    /**
+     * @memberOf Hypergrid.prototype
+     * @param {number} c - grid column index.
+     * @param {string[]} keys
+     */
+    fireSyntheticColumnSortEvent: function(c, keys) {
```

Comment:
We need an API for the user to emit a sort and filtering events (fireSyntheticColumnFilterEvent or the like) as much as we need a cleaner event model. How they user chooses to respond to them was the point of the mixins, I thought.

Maybe you want to go with a model, where the user chooses their own event namespaces (outside of core UI events) and building their own model?